### PR TITLE
Mc logging

### PR DESCRIPTION
**Pull request by glanvl**
Spp-logging module added.
Plus alphabetising.

### CODE REVIEW

File: aggregation_top2_wrangler.py [hunk 1]
```diff
@@ -119,14 +131,16 @@ def lambda_handler(event, context):
...
-                "top2_column": top2_column,
+                "bpm_queue_url": bpm_queue_url,
+                "data": prepared_data,
+                "environment": environment,
                 "run_id": run_id,
-                "bpm_queue_url": bpm_queue_url
+                "survey": survey,
+                "total_columns": total_columns,
+                "top1_column": top1_column,
+                "top2_column": top2_column
```
Comment (Piwington) [inline aggregation_top2_wrangler.py:143]: Top before Total

File: combiner.py [hunk 2]
```diff
@@ -76,17 +75,32 @@ def lambda_handler(event, context):
         aggregated_column = runtime_variables["aggregated_column"]
         aggregation_files = runtime_variables["aggregation_files"]
         bpm_queue_url = runtime_variables["bpm_queue_url"]
+        environment = runtime_variables["environment"]
+        in_file_name = runtime_variables["in_file_name"]
         in_file_name = runtime_variables["in_file_name"]
```
Comment (Piwington) [inline combiner.py:80]: Two lots of in_file_name?

File: dev-requirements.txt [hunk 3]
```diff
@@ -107,3 +108,5 @@ xmltodict==0.12.0
 yamllint==1.20.0
 zipp==0.5.1
 git+https://github.com/ONSdigital/es-functions.git
+git+https://github.com/ONSdigital/spp-logger
+# Force build
```
Comment (Piwington) [inline dev-requirements.txt:112]: I assume this was for an image build. Should go before merge.

File: tests/test_aggregation.py [hunk 4]
```diff
@@ -41,64 +43,81 @@
...
-        "data": None,
-        "total_columns": ["Q608_total"],
-        "additional_aggregated_column": "strata",
         "aggregated_column": "region",
+        "aggregation_type": "sum",
         "cell_total_column": "cell_total",
-        "aggregation_type": "sum"
+        "data": None,
+        "environment": "test - environment",
+        "additional_aggregated_column": "strata",
```
Comment (Piwington) [inline tests/test_aggregation.py:51]: Alphabetical

File: tests/test_aggregation.py [hunk 5]
```diff
@@ -41,64 +43,81 @@
...
 }
 
 method_top2_runtime_variables = {
     "RuntimeVariables": {
-        "data": None,
-        "run_id": "bob",
-        "total_columns": ["Q608_total"],
+        "bpm_queue_url": "fake_queue_url",
         "additional_aggregated_column": "strata",
         "aggregated_column": "region",
```
Comment (Piwington) [inline tests/test_aggregation.py:76]: B after A

File: tests/test_aggregation.py [hunk 6]
```diff
@@ -41,64 +43,81 @@
...
+            "bpm_queue_url": "fake_queue_url",
+            "environment": "test - environment",
+            "factors_parameters": {
+                "RuntimeVariables": {
+                    "region_column": "region",
+                    "regionless_code": "14"
+                }
+            },
+            "incoming_message_group_id": "",
             "in_file_name": "test_wrangler_splitter_input",
```
Comment (Piwington) [inline tests/test_aggregation.py:115]: _ before  letters.

File: aggregation_bricks_splitter_wrangler.py [hunk 7]
```diff
@@ -148,11 +158,13 @@ def lambda_handler(event, context):
 
         payload = {
             "RuntimeVariables": {
+                "bpm_queue_url": bpm_queue_url,
                 "data": json.loads(data_region),
+                "environment": environment,
                 "regionless_code": regionless_code,
                 "region_column": region_column,
```
Comment (Piwington) [inline aggregation_bricks_splitter_wrangler.py:165]: _ before letters.

**Review by Piwington — CHANGES_REQUESTED** [review]
The Alphabet

File: tests/test_aggregation.py (hunk 6 above)
Comment (glanvl) [reply tests/test_aggregation.py:115]: Are you sure?

**Review by rhodriguerrier — APPROVED** [review]

**Review by Piwington — APPROVED** [review]

**Pull request merged**